### ISSUE TITLE
Performance degrades after training

### ISSUE THREAD

**Issue by tilarids** [nilboy/tensorflow-yolo#16]
I've followed the README guide to train this model but after several days of training (1m steps) the resulting performance is sometimes worse than the starting checkpoint. The loss is going down comparing to the first couple steps but doesn't change much after 20000 steps or so. Here is what I have in the end:
```
2017-04-08 14:51:43.589313: step 999950, loss = 3.99 (57.7 examples/sec; 0.277 s
ec/batch)
2017-04-08 14:51:46.701805: step 999960, loss = 8.01 (54.5 examples/sec; 0.294 s
ec/batch)
2017-04-08 14:51:49.754212: step 999970, loss = 7.63 (48.0 examples/sec; 0.333 s
ec/batch)
2017-04-08 14:51:52.851173: step 999980, loss = 4.31 (58.4 examples/sec; 0.274 s
ec/batch)
2017-04-08 14:51:55.847565: step 999990, loss = 4.15 (55.3 examples/sec; 0.289 s
ec/batch)
```
But if you compare the results of running demo with yolo_tiny.ckpt and trained model.ckpt you can see that the result is far from perfect.

yolo_tiny.ckpt:
![cat_out](https://cloud.githubusercontent.com/assets/47031/24949964/5403fe1a-1faa-11e7-9180-df6c9444bbbb.jpg)
trained model.ckpt:
![cat_out2](https://cloud.githubusercontent.com/assets/47031/24949987/6c6bee68-1faa-11e7-81e2-b237b0d74866.jpg)

Anyone else experienced the same behavior?

**Comment by tsminhuang** [comment]
maybe your model is overfitting. 
Add validation function to decide stop training earlier
More detail you can see in YOLO windows port: https://github.com/Jumabek/darknet

**Comment by tilarids** [comment]
@tsminhuang I'd guess that the optimizer is not set properly (e.g., learning rate is too large). I haven't investigated this in more details but I don't think it's the overfitting. For reference, see the loss plot below:
<img width="892" alt="blobresource" src="https://cloud.githubusercontent.com/assets/47031/24992361/09cc4700-205c-11e7-867f-6411c26800ae.png">

My main concern here is not that I can't train the model but that I can't reproduce the results that are advertised in README and I followed the steps precisely.

**Comment by tsminhuang** [comment]
Maybe you can tried to load earlier check point for example 500K to check the result.
Your loss is quite large, but I did not train VOC dataset I am not sure this is normal

**Comment by tilarids** [comment]
Unfortunately the training pipeline removes old model checkpoints so the
earliest one I have is model.ckpt-975000. I don't plan to restart the
training (as I found another YOLO implementation with training that works
as advertised).

And again, I doubt that it's overfitting. It looks like the optimizer is
simply misconfigured so I wonder if anyone was able to train this model
using the code from this repo, starting with the provided checkpoint and
with the data mentioned in the README.

**Comment by nilboy** [comment]
I don't think the performance degrades. if IOU is greater than 0.5, we think it's correct. Maybe the demo image happens to be not good. But the predict is still right.
